File: app/javascript/channels/test_channel.js
import consumer from "channels/consumer"

const testChannel = consumer.subscriptions.create("TestChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Connected")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

window.testChannel = testChannel

File: app/channels/test_channel.rb
class TestChannel < ApplicationCable::Channel
  def subscribed
    stream_from "testing_messages_channel"
    puts "subscribed called"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
    puts "unsubscribed called"
  end

  def receive(data)
    puts "Received: #{data}"
    puts "Current user: #{current_user.inspect}"
  end
end
